gamify-impl/src/main/java/ch/heigvd/gamify/ui/api/endpoints/EventsController.java

```diff
@@ -56,36 +54,30 @@
         .buildAndExpand(entity.getId());
     var rules = ruleRepository.findByEventTypeAndId_App(event.getType(), app.getName());
     for (Rule r : rules) {
-      addEventPoints(event.getUserId(), app, r, r.getPoints());
+      addEventPointAward(event.getUserId(), app, r, r.getPoints());
     }
     return ResponseEntity.created(location.toUri()).build();
   }
 
-  private void addEventPoints(String username, App app, Rule rule, int points) {
-    var userId = userRepository.findByIdEndUser_UserIdAndIdEndUser_App(username, app);
-    if (userId.isEmpty()) {
-      userRepository.save(
-          EndUser.builder()
-              .idEndUser(EndUserIdentifier.builder()
-                  .userId(username)
-                  .app(app).build())
-              .build()
-      );
-    }
-    var realUser = userRepository.findByIdEndUser_UserIdAndIdEndUser_App(username, app).get();
-    var userPoints = userPointsRepository
-        .findByIdEndUserPoints_IdxCategoryAndIdEndUserPoints_IdxUser(
-            rule.getCategory(), realUser);
-    if (userPoints.isPresent()) {
-      userPoints.get().addPoints(points);
-    } else {
-      userPointsRepository.save(
-          EndUserPoints.builder()
-              .idEndUserPoints(EndUserPointsIdentifier.builder()
-                  .idxCategory(rule.getCategory())
-                  .idxUser(realUser).build())
-              .points(points).build()
-      );
-    }
+  private void addEventPointAward(String username, App app, Rule rule, int points) {
+    var endUser = userRepository
+            .findByIdEndUser_UserIdAndIdEndUser_App(username, app)
+            .orElse(userRepository.save(
+                    EndUser.builder()
+                            .idEndUser(
+                                    EndUserIdentifier.builder()
+                                            .userId(username)
+                                            .app(app).build()
+                            )
+                            .build())
+            );
```

Comment:
Formatting.

Reply:
I plan on opening another PR once this one is merged formatting the whole project.